tests/dev/test_check_init_py.py

```diff
@@ -0,0 +1,152 @@
+import subprocess
+import sys
+from pathlib import Path
+from unittest import mock
+
+import pytest
+
+# Import the module we're testing
+sys.path.insert(0, str(Path(__file__).parent.parent))
+from dev.check_init_py import (
+    check_missing_init_files,
+    get_python_directories,
+    get_tracked_python_files,
+    main,
+)
+
+
+def test_get_python_directories():
+    """Test extraction of directories from Python file paths."""
+    python_files = [
+        "mlflow/__init__.py",
+        "mlflow/tracking/client.py",
+        "mlflow/models/model.py",
+        "mlflow/utils/file_utils.py",
+        "mlflow/utils/logging_utils.py",
+    ]
+
+    directories = get_python_directories(python_files)
+    expected = {
+        "mlflow/tracking",
+        "mlflow/models",
+        "mlflow/utils",
+    }
+
+    assert directories == expected
+
+
+def test_get_python_directories_excludes_root():
+    """Test that root mlflow directory is excluded."""
+    python_files = ["mlflow/__init__.py", "mlflow/client.py"]
+    directories = get_python_directories(python_files)
+
+    # Should not include "mlflow" directory itself
+    assert "mlflow" not in directories
+    assert len(directories) == 0
+
+
+def test_check_missing_init_files(tmp_path, monkeypatch):
+    """Test detection of missing __init__.py files."""
+    # Create test directory structure
+    test_dir1 = tmp_path / "mlflow" / "test_package1"
+    test_dir2 = tmp_path / "mlflow" / "test_package2"
+    test_dir1.mkdir(parents=True)
+    test_dir2.mkdir(parents=True)
+
+    # Create __init__.py in one directory but not the other
+    (test_dir1 / "__init__.py").touch()
+    # test_dir2 intentionally missing __init__.py
+
+    # Change to tmpdir to test relative paths
+    monkeypatch.chdir(tmp_path)
```

Comment:
Let's run the script in this directory using `subprocess`, and check the exit code and outputs. That's all we need to test. We don't have to test implement details such as get_python_directories.